src/host/server.js

```diff
@@ -0,0 +1,124 @@
+/// <reference path="types.js" />
+
+import { Readable } from 'stream';
+
+import express from 'express';
+
+import { HostMessage } from './messages.js';
+
+export class Server {
+  constructor(serverOptions) {
+    this.options = serverOptions;
+
+    this.log = this.options.log;
+
+    this._directories = new Map();
+
+    this._app = express();
+    this._app.use('/:directory', this._request.bind(this));
```

Comment:
`use` and `get` do not have the same behavior here. Namely in this case how they manipulate url members of the request object. I used `use` here because it will remove the matching part of the url.